common/lib/xmodule/xmodule/modulestore/search.py

```diff
@@ -104,18 +105,29 @@ def find_path_to_course():
         # (e.g. sequential and videosequence) currently deal with this form of
         # representing nested positions. This needs to happen before jumping to a
         # module nested in more than one positional module will work.
+
         if n > 3:
             position_list = []
+            masquerading_as_student = False
+            if request and request.session.get('masquerade_settings') and \
+                    request.session.get('masquerade_settings')[usage_key.course_key].role == 'student':
```

Comment:
please use  already populated `course_id` variable.

Reply:
Can you check if this doesn't break when we navigate across courses? 
View live from Studio from Course 1 
and
View live from Studio from Course 2